hw/src/trezor.rs

```diff
@@ -0,0 +1,462 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Trezor hardware wallet module. Supports Trezor v1.
+//! See http://doc.satoshilabs.com/trezor-tech/api-protobuf.html
+//! and https://github.com/trezor/trezor-common/blob/master/protob/protocol.md
+//! for protocol details.
+
+use super::{WalletInfo, TransactionInfo, KeyPath};
+
+use bigint::hash::H256;
+use ethkey::{Address, Signature};
+use hidapi;
+use parking_lot::Mutex;
+use protobuf;
+use protobuf::{Message, ProtobufEnum};
+use serde_json;
+use std::cmp::{min, max};
+use std::fmt;
+use std::sync::Arc;
+use std::time::Duration;
+use bigint::prelude::uint::U256;
+
+use trezor_sys::messages::{EthereumAddress, PinMatrixAck, MessageType, EthereumTxRequest, EthereumSignTx, EthereumGetAddress, EthereumTxAck, ButtonAck};
+
+const TREZOR_VID: u16 = 0x534c;
+const TREZOR_PIDS: [u16; 1] = [0x0001]; // Trezor v1, keeping this as an array to leave room for Trezor v2 which is in progress
+const ETH_DERIVATION_PATH: [u32; 4] = [0x8000002C, 0x8000003C, 0x80000000, 0]; // m/44'/60'/0'/0
+const ETC_DERIVATION_PATH: [u32; 4] = [0x8000002C, 0x8000003D, 0x80000000, 0]; // m/44'/61'/0'/0
+
+
+/// Type of message that can be sent to a Trezor device
+#[derive(Debug, Serialize, Deserialize)]
+pub enum TrezorMessageType {
+	/// Get a list of locked devices.
+	#[serde(rename="getDevices")]
+	GetDevices,
```

Comment:
`GetLockedDevices` might be a better name to distinguish it from `hardwareAccounts` that we already have (and where trezor shows up as well after it's unlocked).